### PR TITLE
New collective ops

### PR DESCRIPTION
**Pull request by jeffhammond**
This is the first step to resolving https://github.com/GlobalArrays/ga/issues/52.  I've added the low-level implementations of {logical,bitwise}{and,or} reductions to ARMCI, COMEX and TCGMSG.

### CODE REVIEW

File: armci/tcgmsg/ipcv4.0/globalop.c [hunk 1]
```diff
@@ -23,83 +23,126 @@ extern void free();
...
     }
-  else
-    Error("idoop: unknown operation requested", (long) n);
+  }
+  else {
+    Error("idoop: unknown operation requested", n);
+  }
 }
 
 static void ddoop(n, op, x, work)
```
Comment (jeffdaily) [inline armci/tcgmsg/ipcv4.0/globalop.c]: You changed the declaration from K&R to C89 for idoop but not for ddoop?  Might as well change it here, too.  Also, did you forget to add the new land, lor, band, bor ops to ddoop?

File: armci/tcgmsg/ipcv5.0/globalop.c [hunk 2]
```diff
@@ -41,83 +41,128 @@ void BRDCST_(long *type, void *buf, long *len, long *originator)
...
+    else if (strncmp(op,"absmin",6) == 0) {
+      while(n--) {
+        register double x1 = TCG_ABS(*x), x2 = TCG_ABS(*work);
+        *x = TCG_MIN(x1, x2);
+        x++; work++;
+      }
+    }
+    else {
+      Error("ddoop: unknown operation requested", (long) n);
+    }
```
Comment (jeffdaily) [inline armci/tcgmsg/ipcv5.0/globalop.c]: Are you missing the new operators here?

**Comment by jeffdaily** [comment]
Although I requested changes be made, I think it best to cancel this pull request and instead focus on #63 since it appears to be all of these changes, plus the use of the new operators.  The comments made here for review should instead of applied to #63.

File: armci/tcgmsg/ipcv4.0/globalop.c (hunk 1 above)
Comment (jeffhammond) [inline armci/tcgmsg/ipcv4.0/globalop.c]: I am not aware of any use for logical/bitwise operations on floating-point numbers.

**Comment by jeffdaily** [comment]
Closing this PR since #63 replaces this one.

**Pull request closed without merging**